### PR TITLE
DBZ-1295 Support for 3state snapshot

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1295

### CODE REVIEW

**Comment by jpechane** [comment]
@gunnarmorling This is ready for review - when approved I'll submit PR for Oracle

File: debezium-core/src/main/java/io/debezium/connector/SnapshotRecord.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    LAST;
+
+    public static SnapshotRecord fromSource(Struct source) {
+        if (source.schema().field(AbstractSourceInfo.SNAPSHOT_KEY) != null && io.debezium.data.Enum.LOGICAL_NAME.equals(source.schema().field(AbstractSourceInfo.SNAPSHOT_KEY).schema().name())) {
+            final String snapshotString = source.getString(AbstractSourceInfo.SNAPSHOT_KEY);
+            if (snapshotString != null) {
+                return SnapshotRecord.valueOf(snapshotString.toUpperCase());
+            }
+        }
+        return null;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/connector/SnapshotRecord.java]: Shouldn't it be `FALSE`?

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/connector/SnapshotRecord.java]: @gunnarmorling No, `FALSE` is the default value in the schema and IIRC we agreed we don't want to send `snapshot` for streming phase

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java [hunk 2]
```diff
@@ -268,8 +269,14 @@ protected Instant timestamp() {
     }
 
     @Override
-    protected boolean snapshot() {
-        return snapshot;
+    protected SnapshotRecord snapshot() {
+        if (snapshot) {
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java]: @gunnarmorling Done

File: debezium-core/src/main/java/io/debezium/connector/SnapshotRecord.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/connector/SnapshotRecord.java]: Ok, so I still think it'd be better to express that default behavior in the enum itself and make the fact, that the value for FALSE is omitted in the source info just a local matter of "rendering" the property there. But it's not too important either.

**Pull request merged**